# **DETERMINAN**

adalah ukuran skalar dari sebuah matriks atau selisih antara perkalian elemen-elemen pada diagonal utama dengan perkalian elemen-elemen pada diagonal sekunder. Determinan matriks hanya dapat dicari dengan matriks persegi. Determinan dari matriks A dapat ditulis $det(A)$ atau $|A|$.

## **Sifat - Sifat Determinan :**
1. Determinan dari matrix identitas adalah 1  
contoh :

In [1]:
import numpy as np
a = np.array([[4,2],[3,1]])
b = np.array([[4,1,3],[6,2,5],[8,4,5]])
c = np.array([[1,0],[0,1]])
d = np.array([[1,0,0],[0,1,0],[0,0,1]])
print(a,"\n")
print(b)

[[4 2]
 [3 1]] 

[[4 1 3]
 [6 2 5]
 [8 4 5]]


In [2]:
# a invers
ainv = np.invert (a)
ainv

array([[-5, -3],
       [-4, -2]])

In [3]:
# b invers
binv = np.invert(b)
binv

array([[-5, -2, -4],
       [-7, -3, -6],
       [-9, -5, -6]])

2. Misalkan a dan b adalah matriks bujur sangakar, maka  
det (c) = det (a) * det (b)  
det (I) = det (a) * det (a inv)  


In [4]:
# determinan a
det_a = np.linalg.det(a)
print(f"Determinan A = {det_a}")

Determinan A = -2.0


In [5]:
# determinan a invers
det_ainv = 1 / det_a
det_ainv

-0.5

In [6]:
# determinan b
det_b = np.linalg.det(b)
print(f"Determinan B = {det_b}")

Determinan B = -6.0


In [7]:
# determinan b invers
det_binv = 1 / det_b
det_binv

-0.16666666666666666

In [8]:
hasil = (det_a * det_ainv) * (det_b *det_binv)
print(hasil)

1.0


3. Merubah baris atau kolom dari matriks maka akan merubah tanda dari determinan

In [9]:
e = np.array([[1,4,2],[1,1,1],[3,2,1]])
print(e, "\n")
det_e = round(np.linalg.det(e))
print(f"Determinan E = {det_e}")

[[1 4 2]
 [1 1 1]
 [3 2 1]] 

Determinan E = 5


In [10]:
#merubah baris dari matriks
f = np.array([[1,1,1],[1,4,2],[3,2,1]])
print(f,"\n")
det_f = round(np.linalg.det(f))
print(f"Determinan F = {det_f}")

[[1 1 1]
 [1 4 2]
 [3 2 1]] 

Determinan F = -5


4. Jika baris/kolom merupakan kelipatan dari baris/kolom yang lain maka determinannya adalah 0

In [11]:
# baris 2 merupakan kelipatan dari baris 1
g = np.array([[1,4,2],[2,8,4],[3,2,1]])
print(g, "\n")
det_g = round(np.linalg.det(g))
print(f"Determinan G = {det_g}")

[[1 4 2]
 [2 8 4]
 [3 2 1]] 

Determinan G = 0


In [12]:
# kolom 2 merupakan kelipatan dari kolom 1
g = np.array([[1,4,2],[2,8,3],[1,4,1]])
print(g, "\n")
det_g = round(np.linalg.det(g))
print(f"Determinan G = {det_g}")

[[1 4 2]
 [2 8 3]
 [1 4 1]] 

Determinan G = 0


5. Jika K = Skalar dan Anxn memiliki nxn maka,  
det (KA) = K^n * det(A)

In [13]:
det_KA = (2**3) * 5
print(f"Determinan KA = {det_KA}")

Determinan KA = 40


In [14]:
# Pembuktian
ke = np.array([[2,8,4],[2,2,2],[6,4,2]])
print(ke, "\n")
det_ke = round(np.linalg.det(ke))
print(f"Determinan ke = {det_ke}")

[[2 8 4]
 [2 2 2]
 [6 4 2]] 

Determinan ke = 40


6. Jika T = Skalar dan  
$$
\begin{equation}  B=
\left[ \begin{array}{cccc} ta&tb&tc\\
d&e&f\\
g&h&i \end{array}\right]
\end{equation}
$$   
$$
\begin{equation} A =
\left[ \begin{array}{cccc} a&b&c\\
d&e&f\\
g&h&i \end{array}\right]
\end{equation}
$$   
$det (B) = t * det(A)$

In [15]:
h = np.array([[2,8,4],[1,1,1],[3,2,1]])
print(h, "\n")
det_h = round(np.linalg.det(e))
print(f"Determinan H = {det_h}")

[[2 8 4]
 [1 1 1]
 [3 2 1]] 

Determinan H = 5


## **Matriks Segitiga Atas**
 adalah matriks bujur sangkar yang semua entri dibawah diagonal utama bernilai nol. Berikut merupakan contoh bentuk matriks segitiga atas : 
$$
\begin{equation} A =
\left[ \begin{array}{cccc} d1& . & .\\
0&d2&.\\
0&0&dn \end{array}\right]
\end{equation}
$$  
$det (A) = d1, d2, ... ,dn$  

Determinan dari matriks segitiga atas adalah hasil perkalian dari setiap elemen diagonal

In [16]:
a = np.array([[1,4,2],[0,4,1],[0,0,1]])
print(a, "\n")
det_a = round(np.linalg.det(a))
print(f"Determinan a = {det_a}")

[[1 4 2]
 [0 4 1]
 [0 0 1]] 

Determinan a = 4


7. Melakukan operasi baris elementer terhadap suatu baris tidak akan merubah determinan dan inversnya

In [17]:
a = np.array([[1,4,2],[0,-3,-1],[3,2,1]])
print(a, "\n")
det_a = round(np.linalg.det(a))
print(f"Determinan a = {det_a}")

[[ 1  4  2]
 [ 0 -3 -1]
 [ 3  2  1]] 

Determinan a = 5


8. det (A) = det (A Tranpuse)

In [18]:
e = np.array([[1,4,2],[1,1,1],[3,2,1]])
print(e, "\n")
det_e = round(np.linalg.det(e))
print(f"Determinan e = {det_e}")

[[1 4 2]
 [1 1 1]
 [3 2 1]] 

Determinan e = 5


In [19]:
# determinan e tranpose
e_tran = e.transpose()
print(e_tran,"\n")
det_etran = round(np.linalg.det(e_tran))
print(f"Determinan e tranpose = {det_etran}")

[[1 1 3]
 [4 1 2]
 [2 1 1]] 

Determinan e tranpose = 5


## **Ekspansi Kofaktor**

adalah salah satu metode yang digunakan untuk menghitung determinan dari sebuah matriks. Berikut adalah langkah-langkah umum untuk menghitung determinan menggunakan ekspansi kofaktor:

1. Pilih baris atau kolom tertentu dari matriks yang akan digunakan untuk menghitung determinan. Misalnya, pilih baris pertama atau kolom pertama.

2. Untuk setiap elemen dalam baris atau kolom yang dipilih, hitung kofaktornya. Kofaktor dari elemen \( $a_{ij}$ \) adalah \( $(-1)^{i+j}$ \) dikalikan dengan determinan dari minor \( $M_{ij}$ \), di mana minor \( $M_{ij}$ \) adalah matriks yang dihasilkan dari matriks asli dengan menghapus baris ke-i dan kolom ke-j.

3. Kalikan setiap kofaktor dengan elemen yang bersangkutan, kemudian tambahkan semua hasil perkalian tersebut. Ini akan memberikan nilai determinan dari matriks.


Rumus ekspansi kofaktor :   

$\text{det}(A) = a_{k \times 1} \times C_{k \times 1} + a_{k \times 2} \times C_{k \times 2} + a_{k \times 3} \times C_{k \times 3}$

In [20]:
e = np.array([[4,3,2,1],[1,1,1,1],[2,3,1,2],[4,4,2,1]])
print(e, "\n")
det_e = round(np.linalg.det(e))
print(f"Determinan e = {det_e}")

[[4 3 2 1]
 [1 1 1 1]
 [2 3 1 2]
 [4 4 2 1]] 

Determinan e = 3


In [21]:
e = np.array([[1,0,0,0],[0,1,0,1],[1,0,0,0],[0,1,2,1]])
print(e, "\n")
det_e = round(np.linalg.det(e))
print(f"Determinan e = {det_e}")

[[1 0 0 0]
 [0 1 0 1]
 [1 0 0 0]
 [0 1 2 1]] 

Determinan e = 0


#### Berikut ini merupakan matriks 4 X 4 yang akan dicari determinannya menggunakan metode langsung dan metode manual

In [22]:
e = np.array([[1,0,0,3],[2,7,0,0],[0,6,3,0],[7,3,1,-5]])
print(e, "\n")
det_e = round(np.linalg.det(e))
print(f"Determinan e = {det_e}")

[[ 1  0  0  3]
 [ 2  7  0  0]
 [ 0  6  3  0]
 [ 7  3  1 -5]] 

Determinan e = -528


#### Pembuktian :
$$
\begin{equation} A =
\left[ \begin{array}{cccc} 1&0&0&3\\
2&7&0&0\\
0&6&3&0\\
7&3&1&-5 \end{array}\right]
\end{equation}
$$   

Langkah 1:  
Hitung determinan matriks minor dari baris pertama.

$$\begin{align}
\begin{vmatrix} 7 & 0 & 0 \\ 6 & 3 & 0 \\ 3 & 1 & -5\end{vmatrix}
\end{align}$$

Langkah 2:   
Gunakan aturan ekspansi kofaktor untuk matriks 3x3 dan hitung determinan untuk masing-masing matriks minor 2x2.

$$\begin{align}
& = 7 \times \text{det} \begin{vmatrix} 3 & 0 \\ 1 & -5 \end{vmatrix} - 0 \times \text{det} \begin{vmatrix} 6 & 0 \\ 3 & -5 \end{vmatrix} + 0 \times \text{det} \begin{vmatrix} 6 & 3 \\ 3 & 1 \end{vmatrix} \\
& = 7 \times ((3 \times -5) - (0 \times 1)) - 0 \times ((6 \times (-5)) - (0 \times 3)) + 0 \times ((6 \times 1) - (3 \times 3)) \\
& = 7 \times (-15) - 0 + 0 \\
& = -105
\end{align}$$

Jadi, determinan matriks minor dari baris pertama adalah -105.  

Langkah 3:  
Hitung determinan matriks untuk elemen kedua dan ketiga dari baris pertama (yang bernilai nol). Karena keduanya nol, maka determinannya juga nol.  

Langkah 4:  
Hitung determinan matriks minor dari elemen keempat baris pertama.

$$\begin{align}
\begin{vmatrix} 2 & 7 & 0 \\ 0 & 6 & 3 \\ 7 & 3 & 1\end{vmatrix}
\end{align}$$

Langkah 5:  
Hitung determinan matriks minor dari elemen keempat baris pertama dan gunakan aturan ekspansi kofaktor untuk matriks 3x3.   

$$\begin{align}
& = 2 \times \text{det} \begin{vmatrix} 6 & 3 \\ 3 & 1 \end{vmatrix} - 7 \times \text{det} \begin{vmatrix} 0 & 3 \\ 7 & 1 \end{vmatrix} + 0 \times \text{det} \begin{vmatrix} 0 & 6 \\ 7 & 3 \end{vmatrix} \\
& = 2 \times ((6 \times 1) - (3 \times 3)) - 7 \times ((0 \times 1) - (3 \times 7)) + 0\\
& = 2 \times (-3) - 7 \times (-21) \\
& = -6 + 147 \\
& = 141
\end{align}$$

Jadi, determinan matriks minor dari elemen keempat baris pertama adalah 141.

Langkah 6:  
Hitung determinan keseluruhan dengan mengalikan masing-masing determinan matriks minor dengan kofaktornya, yaitu (-1)^(i+j), di mana i dan j adalah indeks baris dan kolom elemen tersebut.

$$\begin{align}
&\text{det(A)} = 1 \times (-105) - 0 \times 0 + 0 \times 0 - 3 \times 141\\
&\text{det(A)} = -105 - 0 + 0 - 423 \\
&\text{det(A)} = -528
\end{align}$$

Jadi, determinan dari matriks tersebut adalah -528.